tests/testsuite/test_integration_with_browsers.py

```diff
@@ -0,0 +1,287 @@
+import os
+import splinter
+import pytest
+
+DOMAIN = 'liquid.example.org'
+URL = 'http://'+DOMAIN
+
+ADMIN_USERNAME = 'testadmin'
+ADMIN_PASSWORD = 'test-liquid.example.org'
+HOTSPOT_SSID = 'testhotspotssid'
+HOTSPOT_PASSWORD = 'testhotspotpassword'
+
+APP_NAMES = [
+    "Hoover",
+    "Hypothesis",
+    "DokuWiki",
+    "Matrix",
+    "Davros",
+]
+
+BROWSERS = [
+    'firefox',
+    'chrome',
+]
+
+BROWSER_OPTS = {
+    'firefox': {
+    },
+    'chrome': {
+        'service_args': ['--verbose', '--log-path=chromedriver.log'],
+    },
+}
+
+
+WELCOME_FILE_PATH = '/var/lib/liquid/core/welcome_done'
+
+def skip_if_welcome_not_set():
+    if not os.path.isfile(WELCOME_FILE_PATH):
```

Comment:
The concept is sound, but! you're testing the system as a black box over HTTP. You should check for welcome screen via HTTP, e.g. by doing a GET on `/` and checking if it redirects to `/welcome`.